source/fuzz/transformation_changing_memory_scope.cpp

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) 2021 Mostafa Ashraf
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "source/fuzz/transformation_changing_memory_scope.h"
+
+#include "source/fuzz/fuzzer_util.h"
+#include "source/fuzz/instruction_descriptor.h"
+
+namespace spvtools {
+namespace fuzz {
+
+TransformationChangingMemoryScope::TransformationChangingMemoryScope(
+    protobufs::TransformationChangingMemoryScope message)
+    : message_(std::move(message)) {}
+
+TransformationChangingMemoryScope::TransformationChangingMemoryScope(
+    const protobufs::InstructionDescriptor& needed_instruction,
+    uint32_t memory_scope_new_value_id) {
+  *message_.mutable_needed_instruction() = needed_instruction;
+  message_.set_memory_scope_new_value_id(memory_scope_new_value_id);
+}
+
+bool TransformationChangingMemoryScope::IsApplicable(
+    opt::IRContext* ir_context, const TransformationContext& /*unused*/) const {
+  // The instruction must be exist.
+  auto needed_instruction =
+      FindInstruction(message_.needed_instruction(), ir_context);
+  if (!needed_instruction) {
+    return false;
+  }
+
+  // Instruction must be atomic instruction only.
+  if (!IsAtomicInstruction(needed_instruction->opcode())) {
+    return false;
+  }
```

Comment:
See your other PR and follow the same approach. We do not need to check the opcode in IsApplicable.

Reply:
Done.